test/GPv2Settlement.test.ts

```diff
@@ -239,4 +240,38 @@ describe("GPv2Settlement", () => {
       ).to.deep.equal(ethers.constants.Zero);
     });
   });
+
+  describe("transferBalanceTo", () => {
+    it("allows owner to transfer balance of multiple tokens out of settlement contract", async () => {
+      const tokens: Contract[] = [
+        await waffle.deployContract(deployer, ERC20, ["token0", "18"]),
+        await waffle.deployContract(deployer, ERC20, ["token1", "18"]),
```

Comment:
Can we use waffle's mock contracts here instead of real implementations?

Reply:
I guess... but why?

Reply:
Because in unit tests it's good practice to use mocks instead of "real objects" as your dependencies. This allows to isolate the behaviour you are trying to test and set clear instructions on what your dependencies are expected to do and how your object under test should react. If you test multiple components together, you are likely writing an integration test.

While for ERC20 it might be easy enough to use a fake implementation to get the desired behavior of our dependencies, this is not the case for more complex dependencies and as such it's more consistent to also mock ERC20 (e.g. as we do in dex-contracts).

Reply:
Isn't it just going be more setup to create a mock contract and tell it what it is supposed to do at every step of the way instead of just having the real thing. Seems to me like the real thing is always better and (in this case) overall shorter to setup. I will look into this in the morning, perhaps I just don't get it.

Reply:
One more argument is that we probably want to have tests to make sure this works with non-standard ERC20 implementations (such as transfer reverting vs returning false). If we use the OpenZeppelin implementation for tests, thats not really possible.

Reply:
Real objects require future maintainers of the test to understand how the inner logic of the dependency works in order to understand the setup of the test (which in the case of ERC20 can be assumed trivial but in more complicated cases is not). Setting up mocks should be as easy as saying `if this method gets called, return this`. It can even be as easy as sysing `all calls should pass` (e.g. return true for ERC20).